### PR TITLE
[Enhancement] Support detect edit log port connectivity

### PR DESCRIPTION
**Pull request by nshangyiming**
Fixes SR-21413

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/system/PortConnectivityChecker.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ */
+public class PortConnectivityChecker extends LeaderDaemon {
+    private static final Logger LOG = LogManager.getLogger(PortConnectivityChecker.class);
+    private final List<Integer> portsToCheck = new ArrayList<>();
+    private final ExecutorService executor;
+    private Map<Pair<String, Integer>, Boolean> currentPortStates = new HashMap<>();
+
+    public PortConnectivityChecker() {
+        super("PortConnectivityChecker");
+        portsToCheck.add(Config.edit_log_port);
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/system/PortConnectivityChecker.java:69]: The edit_log_port and rpc_port can be different in FE nodes. You should get the value from frontend object.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 50 / 54 (92.59%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/PortConnectivityChecker.java|45|49|91.84%|[105, 106, 122, 123]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 57 / 61 (93.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/PortConnectivityChecker.java|50|54|92.59%|[112, 113, 129, 130]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/Frontend.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request merged**